Added ability to toggle display of scalars

soneymathew opened the pull request:
When using voyager for understanding larger graphs it's sometimes useful to hide the scalars. Please let me know your thoughts on this change.

Added an option to toggle display of leaf fields
![image](https://user-images.githubusercontent.com/1210516/44308283-dde6b700-a3f5-11e8-9992-14fef230ad1c.png)

When leaf fields are hidden it becomes a bit more easier to navigate and understand the connection between the edges.
![image](https://user-images.githubusercontent.com/1210516/44308291-0ec6ec00-a3f6-11e8-8510-4fa4188121c8.png)

As part of this change the displayOptions are accessible within the dot generation logic, and we could take this a step further and extend with much more awesome display options as well.

IvanGoncharov commented:
@soneymathew Wow, great PR 👍

IvanGoncharov commented on src/graph/dot.ts:
```diff
@@ -1,11 +1,11 @@
+export const getDotSelector = createSelector(getTypeGraphSelector, getDisplayOptions, getDot);
 
-function getDot(typeGraph): string {
+function getDot(typeGraph, displayOptions): string {
```
@soneymathew IMO Passing `displayOptions` between function adds a lot of boilerplate. Can you move all functions(except `objectValues`) inside `getDot`:
```js
function getDot() {

  return (/* ... */);

  function nodeField() {
    // can access options
  }
}

soneymathew replied:
Thanks for the feedback, will update

soneymathew replied:
Thanks for the feedback, will update

soneymathew replied:
done

IvanGoncharov commented on src/graph/dot.ts:
```diff
@@ -80,17 +82,24 @@ function nodeLabel(node) {
   }</FONT><BR/>${kindLabel}</TD>
       </TR>
-      ${objectValues(node.fields, nodeField)}
+      ${objectValues(node.fields, nodeField, displayOptions)}
```
@soneymathew Can you make the third argument a filter predicate and pass `canDisplayRow`?

IvanGoncharov commented on src/graph/dot.ts:
```diff
@@ -80,17 +82,24 @@ function nodeLabel(node) {
 
-function nodeField(field) {
+function canDisplayRow(type, displayOptions) {
+  if(type.kind === 'SCALAR') {
```
@soneymathew Scalar doesn't include enums. Do you want to skip only scalars or any leaf fields (Scalars + Enums)?

IvanGoncharov commented:
@soneymathew Great feature 🎉 
I add a few code style comment but overall it looks great.
One big issue for me is "Display Scalars" name since I would expect such option to show/hide scalar types not a fields with scalars.
How about "Show leafs" or "Show leaf fields"?

soneymathew commented:
That makes sense, I will change to Show leafs ? Should that hide the fields that connect to other types? I initially implemented it to hide all the things except type name, then I decided at the time to retain the display of fields that connected to other types.

IvanGoncharov commented:
@soneymathew Just to be sure we both thinking about the same feature.
Idea is to remove everything except edges and elements associated with those edges, Right? 
End goal: type boxes become shorter and more types fit on the screen \+ it helps to focus on the connection between types.

> That makes sense, I will change to Show leafs ?

I checked GraphQL spec and it looks like "Leaf field" is correct term:
http://facebook.github.io/graphql/June2018/#sec-Leaf-Field-Selections

\+ In that context `leaf` could mean two things:
1. type that doesn't reference other types
2. field that doesn't reference type on graphl (composite type).

But `leaf field` can mean only `#2`

IvanGoncharov commented:
> Not doing this, or should we add a new display option ?

IMHO, for big schemas, it will look like a total mess so I don't think we need it.

soneymathew commented:
@IvanGoncharov I incorporated the feedbacks and updated the main comment with new images, please have a look and let me know if you need any more changes. thanks for the very quick feedbacks

IvanGoncharov commented:
@soneymathew Merged 🎉 
Disclaimer: This repo stuck in the middle of big refactoring so I can't release new version since some of functionality is broken.
Don't have a good ETA for release but I will try to do it this month.
